From f00912f0628aff15e882bdc1619f5de1ddf5c720 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Enrique=20Oca=C3=B1a=20Gonz=C3=A1lez?= <eocanha@igalia.com>
Date: Tue, 17 Nov 2015 16:51:27 +0000
Subject: [PATCH] Properly handle drain requests while flushing

Without this commit the decoder streaming thread stops without ever attending
the drain request, leaving the decoder input thread waiting forever.

---
 omx/gstomx.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/omx/gstomx.c b/omx/gstomx.c
index 45732c3..784a5d7 100644
--- a/omx/gstomx.c
+++ b/omx/gstomx.c
@@ -737,6 +737,13 @@ gst_omx_component_new (GstObject * parent, const gchar * core_name,
 
   g_mutex_lock (&comp->lock);
   gst_omx_component_handle_messages (comp);
+
+  if (err != OMX_ErrorNone && comp->last_error == OMX_ErrorNone) {
+    GST_ERROR_OBJECT (comp->parent,
+        "Last operation returned an error. Setting last_error manually.");
+    comp->last_error = err;
+  }
+
   g_mutex_unlock (&comp->lock);
 
   return comp;
